### PR TITLE
Add configuration option to cache fetched URIs in the executor so tha…

### PR DESCRIPTION
**Pull request by verma7**
…t rolling restarts/failed task restarts do not have to depend on the source artifactory/S3 being up

### CODE REVIEW

**Comment by zhiyanshao** [comment]
@verma7  When will the cached file be evicted or refreshed? Will it always fetch from artifactory/S3 if they are available?

**Comment by verma7** [comment]
@zhiyanshao 
> When will the cached file be evicted or refreshed? 

The cached file will be evicted only when the cache is full.

> Will it always fetch from artifactory/S3 if they are available?

No, it will fetch it once and then cache it for as long as possible. We need to change the version number of the url for it to fetch a different binary.

You can read more details here: http://mesos.apache.org/documentation/latest/fetcher/ and  http://mesos.apache.org/documentation/latest/fetcher-cache-internals/

File: cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java [hunk 1]
```diff
@@ -56,7 +58,8 @@ public static ExecutorConfig create(
             @JsonProperty("executor_location") String executorLocation,
             @JsonProperty("cassandra_location") String cassandraLocation,
             @JsonProperty("libmesos_location") String libmesosLocation,
-            @JsonProperty("emc_ecs_workaround") boolean emcEcsWorkaround)
+            @JsonProperty("emc_ecs_workaround") boolean emcEcsWorkaround,
+            @JsonProperty("cache_fetched_uris") boolean cacheFetchedUris)
```
Comment (szhou1234) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java]: nit: we don't cache the URI, but the binaries.

Comment (verma7) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java]: We fetch the URIs and then cache the fetched URIs. I am open to suggestions for better names.

Comment (mohitsoni) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java]: `cache_fetched_uris` sounds good to me. @szhou1234 Is there any better name that you can suggest ?

Comment (szhou1234) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java]: Not a big deal. Maybe cache_binary or cache_artifactory?

Comment (verma7) [inline cassandra-commons/src/main/java/com/mesosphere/dcos/cassandra/common/config/ExecutorConfig.java]: I am inclined to leave it as-is. Thanks!

**Pull request merged**